go/vt/vttablet/grpctmclient/client.go

```diff
@@ -92,14 +103,17 @@ type tmc struct {
 	client tabletmanagerservicepb.TabletManagerClient
 }
 
+// rpcClientMap maps an address to a tmc
+type rpcClientMap map[string](chan *tmc)
+
 // grpcClient implements both dialer and poolDialer.
 type grpcClient struct {
 	// This cache of connections is to maximize QPS for ExecuteFetchAs{Dba,App},
 	// CheckThrottler and FullStatus. Note we'll keep the clients open and close them upon Close() only.
 	// But that's OK because usually the tasks that use them are one-purpose only.
 	// The map is protected by the mutex.
-	mu           sync.Mutex
-	rpcClientMap map[string]chan *tmc
+	mu            sync.Mutex
+	rpcClientMaps map[DialPoolGroup]rpcClientMap
```

Comment:
`DialPoolGroupThrottler` and `DialPoolGroupVTOrc` do not need a channel and should only open One connection per tablet address.
better to have a different type for them and leave the other `concurrent one` seperate.

Reply:
Is it OK if I still make the code idential for all of them, but use channel size `1` for VTOrc and Throttler?

Reply:
See proposed solution in https://github.com/vitessio/vitess/pull/15560/commits/a2dd6fee850cc5a405414d0a190e67703dcd6623

Reply:
CI doesn't like https://github.com/vitessio/vitess/commit/a2dd6fee850cc5a405414d0a190e67703dcd6623